modules/javafx.base/src/main/java/com/sun/javafx/PlatformUtil.java

```diff
@@ -238,8 +238,7 @@ private static File getRTDir() {
             // Strip everything after the last "/" or "\" to get rid of the jar filename
             int lastIndexOfSlash = Math.max(
                     s.lastIndexOf('/'), s.lastIndexOf('\\'));
-            return new File(new URL(s.substring(0, lastIndexOfSlash + 1)).getPath())
-                    .getParentFile();
+            return new File(new URL(s.substring(0, lastIndexOfSlash + 1)).getPath());
```

Comment:
The previous code looks like a hold-over from JDK 8, where jfxrt.jar was in `lib/ext`. It also assumes that the JavaFX runtime is being loaded from a jar URL, which isn't necessarily the case. Probably the only reason it hasn't caused problems before now is that it is only used to locate `javafx.platform.properties`. Worth noting is that we won't get here in the case JavaFX is jlinked into the Java runtime, although in that case, the fallback method of locating it relative to the JDK should be used.

I'll take a closer look this specific change, but I think it should be OK.

I'll defer the review as a whole to Johan.

Reply:
I can confirm that this part of the fix is correct. The change to DPI looks correct to me, too.